exporter/awsemfexporter/metric_translator.go

```diff
@@ -421,3 +468,221 @@ func needsCalculateRate(pmd *pdata.Metric) bool {
+		labelsList := make([][]string, 0)
+		metricsList := make([]map[string]string, 0)
+		fieldMap["Namespace"] = v.Namespace
+
+		lList := make([]string, 0)
+		for key, val := range v.Labels {
+			lList = append(lList,key)
+			fieldMap[key] = val
+		}
+
+		labelsList = append(labelsList, lList)
+
+		for key, val := range v.Metrics {
+			metricDef := make(map[string]string)
+			metricDef["Name"] = key
+			metricDef["Unit"] = val.Unit
+			metricsList = append(metricsList, metricDef)
+			fieldMap[key] = val.Value
+		}
+
+		cwm := &CWMeasurement {
+			Namespace: v.Namespace,
+			Dimensions: labelsList,
+			Metrics: metricsList,
+		}
+
+		cwmMap := make(map[string]interface{})
+		cwmMap["CloudWatchMetrics"] = cwm
+		cwmMap["Timestamp"] = v.Timestamp
+		fieldMap["_aws"] = cwmMap
+
+		pleMsg, err := json.Marshal(fieldMap)
+		if err != nil {
+			continue
+		}
+		metricCreationTime := v.Timestamp
+
+		logEvent := NewLogEvent(
+			metricCreationTime,
+			string(pleMsg),
+		)
+		logEvent.LogGeneratedTime = time.Unix(0, metricCreationTime*int64(time.Millisecond))
+		ples = append(ples, logEvent)
+	}
+	return ples
+}
+
+// TranslateOtToCWMetric converts OT metrics to CloudWatch Metric format
+func TranslateOtToCWMetric(rm *pdata.ResourceMetrics, dimensionRollupOption string, namespace string) ([]*CWMetrics, int) {
+	var cwMetricList []*CWMetrics
+	totalDroppedMetrics := 0
+	var instrumentationLibName string
+
+	if len(namespace) == 0 && !rm.Resource().IsNil() {
+		serviceName, svcNameOk := rm.Resource().Attributes().Get(conventions.AttributeServiceName)
+		serviceNamespace, svcNsOk := rm.Resource().Attributes().Get(conventions.AttributeServiceNamespace)
+		if svcNameOk && svcNsOk && serviceName.Type() == pdata.AttributeValueSTRING && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = fmt.Sprintf("%s/%s", serviceNamespace.StringVal(), serviceName.StringVal())
+		} else if svcNameOk && serviceName.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceName.StringVal()
+		} else if svcNsOk && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceNamespace.StringVal()
+		}
+	}
+
+	if len(namespace) == 0 {
+		namespace = defaultNameSpace
+	}
+
+	ilms := rm.InstrumentationLibraryMetrics()
+	for j := 0; j < ilms.Len(); j++ {
+		ilm := ilms.At(j)
+		if ilm.IsNil() {
+			continue
+		}
+		if ilm.InstrumentationLibrary().IsNil() {
+			instrumentationLibName = noInstrumentationLibraryName
+		} else {
+			instrumentationLibName = ilm.InstrumentationLibrary().Name()
+		}
+
+		metrics := ilm.Metrics()
+		for k := 0; k < metrics.Len(); k++ {
+			metric := metrics.At(k)
+			if metric.IsNil() {
+				totalDroppedMetrics++
+				continue
+			}
+			cwMetrics := getCWMetrics(&metric, namespace, instrumentationLibName, dimensionRollupOption)
+			cwMetricList = append(cwMetricList, cwMetrics...)
+		}
+	}
+	return cwMetricList, totalDroppedMetrics
+}
+
+func TranslateCWMetricToEMF(cwMetricLists []*CWMetrics) []*LogEvent {
```

Comment:
Likewise for this, can you replace this above so that git won't think that this is a new change?